config-check/src/main/java/com/okta/commons/configcheck/ConfigurationValidator.java

```diff
@@ -42,75 +42,128 @@ private ConfigurationValidator() {}
      * @param clientId The Client Id to be validated
      */
-    public static void validateClientId(String clientId) {
-        hasText(clientId, ERRORS.getString("clientId.missing"));
-        doesNotContain(clientId, "{clientId}",  ERRORS.getString("clientId.containsBrackets"));
+    public static void assertClientId(String clientId) {
+        validateClientId(clientId).ifInvalidThrow();
+    }
+
+    public static ValidationResponse validateClientId(String clientId) {
+        ValidationResponse response = new ValidationResponse();
+        if (!hasText(clientId)) {
+            response.setMessage(ERRORS.getString("clientId.missing"));
+        } else if (contains(clientId, "{clientId}")) {
+            response.setMessage(ERRORS.getString("clientId.containsBrackets"));
+        }
+        return response;
     }
 
     /**
      * Validates that a client secret is not null and contains does not contain the string {@code {clientSecret}}.
      * @param clientSecret the Client Secret to be validated
      */
-    public static void validateClientSecret(String clientSecret) {
-        hasText(clientSecret, ERRORS.getString("clientSecret.missing"));
-        doesNotContain(clientSecret, "{clientSecret}",  ERRORS.getString("clientSecret.containsBrackets"));
+    public static void assertClientSecret(String clientSecret) {
+        validateClientSecret(clientSecret).ifInvalidThrow();
+    }
+
+    public static ValidationResponse validateClientSecret(String clientSecret) {
+        ValidationResponse response = new ValidationResponse();
+        if (!hasText(clientSecret)) {
+            response.setMessage(ERRORS.getString("clientSecret.missing"));
+        } else if (contains(clientSecret, "{clientSecret}")) {
+            response.setMessage(ERRORS.getString("clientSecret.containsBrackets"));
+        }
+        return response;
     }
 
     private static String formattedErrorMessage(String messageKey, Object... args) {
         String message = ERRORS.getString(messageKey);
         return MessageFormat.format(message, args);
     }
 
+    private static ValidationResponse validateHttpsUrl(String url, String keyPrefix) {
+        ValidationResponse response = new ValidationResponse();
+        if (!hasText(url)) {
+            response.setMessage(ERRORS.getString(keyPrefix + ".missing"));
+        } else if (contains(url, "{yourOktaDomain}")) {
+            response.setMessage(ERRORS.getString(keyPrefix + ".containsBrackets"));
+        } else {
+            try {
+                URL tempUrl = new URL(url);
+                if (!"https".equalsIgnoreCase(tempUrl.getProtocol())) {
+                    response.setMessage(formattedErrorMessage(keyPrefix + ".nonHttpsInvalid", url));
+                } else if (tempUrl.getHost().endsWith(".com.com")){
+                    response.setMessage(formattedErrorMessage(keyPrefix + ".invalid", url));
+                } else if (tempUrl.getHost().endsWith("-admin.okta.com")
```

Comment:
gross. brittle. ;) What're we gonna do when we have `okta-apac.com`. Maybe this should be a regex?

Reply:
Yeah, I agree, the goal is to catch specific common error of somebody copy/pasting an `-admin` url from their browser, but not catch other _valid_ cases like: `custom-admin-domain.example.com`

Suggestions?